cmake/IntelDFP.cmake

```diff
@@ -0,0 +1,396 @@
+
+include (FetchContent)
+find_program (GIT_EXECUTABLE git)
+
+set (_default_url "https://netlib.org/misc/intel/IntelRDFPMathLib20U2.tar.gz")
```

Comment:
A build failure on the C driver was observed related to downloading this tarball:

https://spruce.mongodb.com/task/mongo_c_driver_arm_ubuntu1804_debug_compile_sasl_openssl_cse_patch_a8560d7488778e480b7c61a0f8931ddbcc310c7d_63c6c96fd6d80a4e2436bcd1_23_01_17_16_14_40/logs?execution=1

```
[2023/01/17 16:25:00.937] -- Using src='https://netlib.org/misc/intel/IntelRDFPMathLib20U2.tar.gz';
[2023/01/17 16:25:00.937] CMake Error at intel_dfp-subbuild/intel_dfp-populate-prefix/src/intel_dfp-populate-stamp/download-intel_dfp-populate.cmake:170 (message):
[2023/01/17 16:25:00.937]   Each download failed!
[2023/01/17 16:25:00.937]     error: downloading 'https://netlib.org/misc/intel/IntelRDFPMathLib20U2.tar.gz'; failed
[2023/01/17 16:25:00.937]           status_code: 28
[2023/01/17 16:25:00.937]           status_string: "Timeout was reached"
[2023/01/17 16:25:00.937]           log:
[2023/01/17 16:25:00.937]           --- LOG BEGIN ---
[2023/01/17 16:25:00.937]             Trying 160.36.131.221:443...
[2023/01/17 16:25:00.937]   connect to 160.36.131.221 port 443 failed: Connection timed out
[2023/01/17 16:25:00.937]   Failed to connect to netlib.org port 443 after 129521 ms: Couldn't connect
[2023/01/17 16:25:00.937]   to server
[2023/01/17 16:25:00.937]   Closing connection 0
[2023/01/17 16:25:00.937]           --- LOG END ---
[2023/01/17 16:25:00.937] CMakeFiles/intel_dfp-populate.dir/build.make:99: recipe for target 'intel_dfp-populate-prefix/src/intel_dfp-populate-stamp/intel_dfp-populate-download' failed
[2023/01/17 16:25:00.937] make[2]: *** [intel_dfp-populate-prefix/src/intel_dfp-populate-stamp/intel_dfp-populate-download] Error 1
[2023/01/17 16:25:00.937] CMakeFiles/Makefile2:82: recipe for target 'CMakeFiles/intel_dfp-populate.dir/all' failed
[2023/01/17 16:25:00.937] make[1]: *** [CMakeFiles/intel_dfp-populate.dir/all] Error 2
[2023/01/17 16:25:00.937] Makefile:90: recipe for target 'all' failed
[2023/01/17 16:25:00.937] make: *** [all] Error 2
[2023/01/17 16:25:00.937] CMake Error at /home/ubuntu/.cache/libmongocrypt/build.1/cmake/3.25.1/share/cmake-3.25/Modules/FetchContent.cmake:1616 (message):
[2023/01/17 16:25:00.937]   Build step for intel_dfp failed: 2
[2023/01/17 16:25:00.937] Call Stack (most recent call first):
[2023/01/17 16:25:00.937]   /home/ubuntu/.cache/libmongocrypt/build.1/cmake/3.25.1/share/cmake-3.25/Modules/FetchContent.cmake:1756:EVAL:2 (__FetchContent_directPopulate)
[2023/01/17 16:25:00.937]   /home/ubuntu/.cache/libmongocrypt/build.1/cmake/3.25.1/share/cmake-3.25/Modules/FetchContent.cmake:1756 (cmake_language)
[2023/01/17 16:25:00.937]   cmake/IntelDFP.cmake:46 (FetchContent_Populate)
[2023/01/17 16:25:00.937]   CMakeLists.txt:28 (include)
```

A subsequent run on that variant passed. If the download may be unreliable, can the source be copied instead of downloaded?